x/btcstaking/keeper/msg_server.go

```diff
@@ -295,6 +294,13 @@ func (ms msgServer) CreateBTCDelegation(goCtx context.Context, req *types.MsgCre
 	if err := ms.AddBTCDelegation(ctx, newBTCDel); err != nil {
```

Comment:
Here we are adding the BTC delegation normally as it is the same for BTC delegations to Babylon vs Consumers, that means for both cases events `EventBTCDelegationStateUpdate` are being added and later processed in `ProcessAllPowerDistUpdateEvents` - https://github.com/babylonlabs-io/babylon/blob/be9eb6b9f4db544eb35db8b4767852d12910b9eb/x/finality/keeper/power_dist_change.go#L170

Since rewards of consumer chains will not be handled by the Babylon chain we should handle and do not store the changes of those finality providers in the incentive store (do not call `processRewardTracker` https://github.com/babylonlabs-io/babylon/blob/be9eb6b9f4db544eb35db8b4767852d12910b9eb/x/finality/keeper/power_dist_change.go#L216) for consumers BTC delegations and FPs

Reply:
We can do this because the protocol requires that every BTC delegation has to at least mult-stake to a Babylon FP ([ref](https://github.com/babylonlabs-io/babylon/blob/1a0f10f4f7918a5b792a42dcc1b18c6b13c17d73/x/btcstaking/keeper/btc_delegations.go#L261-L264)). This is a product decision in order to protect Babylon value proposition.

Reply:
yeap, but we would be tracking for multiple fps https://github.com/babylonlabs-io/babylon/blob/be9eb6b9f4db544eb35db8b4767852d12910b9eb/x/finality/keeper/power_dist_change.go#L422
while the only ones that matter for the incentives reward tracker are the FPs from Babylon, not from consumers

Reply:
Then this means the `processRewardTracker` function or the new lazy reward protocol is not compatible with multi-staking since it assumes all FPs are Babylon FPs. In this case wdyt we modify `processRewardTracker` and `ProcessAllPowerDistUpdateEvents` to only handle Babylon FPs, like here https://github.com/babylonlabs-io/babylon/pull/424/commits/1e825db167b72efc5468fff24799606bf4ef91e2?

Reply:
sounds good to me